### ISSUE TITLE
Core clock can't go higher than than 935Mhz, AMD 5700

### ISSUE THREAD

**Issue by jbaevea** [azeam/powerupp#40]
I am new to powerupp and I don't know whether there is something I did wrong. Below is a screen shot of the settings using powerupp.

![Screenshot from 2022-06-06 11-37-30](https://user-images.githubusercontent.com/106995931/172242219-1a60b89c-6fdd-41c2-86cc-72a6f70aafc3.png)
 
And below is the content of pp_od_clk_voltage when the above setting is active.

![Screenshot from 2022-06-06 11-30-36](https://user-images.githubusercontent.com/106995931/172242270-5090be29-7cec-4049-9c00-c228e0b90ce7.png)
 
I thought power state 1: in OD_SCLK should be 1850 instead of 935. Is my assumption correct?

The GPU is Power Color AXRX 5700. OS is Ubuntu 20.04, core version 5.13. AMDGPU version 21.40.1. Below are screen captures of settings using MorePowerTool on Windows.

![Morepowertool - Frequency](https://user-images.githubusercontent.com/106995931/172242343-8283b192-0a14-4e4e-800d-a736e49321a4.JPG)
![Morepowertool - overdrive limit](https://user-images.githubusercontent.com/106995931/172242345-7426a8c1-caf2-44ca-bac9-321afe548871.JPG)
![Morepowertool - Power and voltage](https://user-images.githubusercontent.com/106995931/172242346-8de47ad8-fdb1-4199-b57d-c879797b4bc4.JPG)

**Comment by azeam** [comment]
The `Max Gfx voltage` setting seems very low, have you tried setting it to 1100-1200 mV?

**Comment by jbaevea** [comment]
Yes, it is used by the new R-mode for Navi and Big Navi cards from Team Red Miner. 1025Mhz is enough and is working well.

**Comment by azeam** [comment]
And it still won't reach higher frequencies? Does it change if you put it under some load?

**Comment by jbaevea** [comment]
I can't get powerupp to use core clk 1025Mhz. As I showed the content of file pp_od_clk_voltage, the core clock is capped at 935Mhz. So I started using corectrl and it can set the core clk at 1025 correctly. See the content of file pp_od_clk_voltage here. Pay attention to the value of line 1: in OD_SCLK block. Somehow I feel that upp is not setting the pp table correctly for this configuration.  Note that I lower down the memory clock a bit for stability.  

OD_SCLK:
0: 1025Mhz
1: 1025Mhz
OD_MCLK:
1: 908MHz
OD_VDDC_CURVE:
0: 809MHz 685mV
1: 890MHz 685mV
2: 1030MHz 685mV
OD_RANGE:
SCLK:     800Mhz       1850Mhz
MCLK:     625Mhz        930Mhz
VDDC_CURVE_SCLK[0]:     800Mhz       1850Mhz
VDDC_CURVE_VOLT[0]:     600mV        1200mV
VDDC_CURVE_SCLK[1]:     800Mhz       1850Mhz
VDDC_CURVE_VOLT[1]:     600mV        1200mV
VDDC_CURVE_SCLK[2]:     800Mhz       1850Mhz
VDDC_CURVE_VOLT[2]:     600mV        1200mV
